src/enemy/eyegore.c

```diff
@@ -0,0 +1,485 @@
+        }
+    }
+}
+
+void Eyegore_Action4(EyegoreEntity* this) {
+    GetNextFrame(super);
+    switch (super->frame & 7) {
+        case 1:
+            sub_08031024(this);
+            sub_08030E3C(this);
+            break;
+        case 2:
+            if ((super->frame & 0x80) != 0) {
+                super->animationState = this->unk_7f;
+                InitializeAnimation(super, super->animationState + 4);
+            }
+            if (this->unk_7a != 0) {
+                sub_08031250(this);
+            }
+            if (GravityUpdate(super, 0x1c00) == 0) {
+                sub_08030E70(this);
+            }
+
+            break;
+        default:
+            if ((super->frame & 0x80) != 0) {
+                InitializeAnimation(super, super->animationState);
+                if ((this->unk_79 & 0x80) != 0) {
+                    super->action = 5;
+                    this->unk_7a = 0;
+                }
+            }
+            break;
+    }
+}
+
+void Eyegore_Action5(EyegoreEntity* this) {
+    GetNextFrame(super);
+    switch (super->frame & 7) {
+        case 1:
+            if (super->animationState != 2) {
+                if (super->animationState >= 3) {
+                    this->unk_7f = (super->animationState - 1) & 3;
+                } else {
+                    this->unk_7f = (super->animationState + 1) & 3;
+                }
+            }
+            sub_08030E3C(this);
+            break;
+        case 2:
+            if ((super->frame & 0x80) != 0) {
+                super->animationState = this->unk_7f;
+                InitializeAnimation(super, super->animationState + 4);
+            }
+            if (GravityUpdate(super, 0x1c00) == 0) {
+                sub_08030E70(this);
+            }
+            break;
+        default:
+            if ((super->frame & 0x80) != 0) {
+                InitializeAnimation(super, super->animationState);
+                if (sub_08049FDC(super, 1)) {
+                    sub_08030E58(this);
+                } else {
+                    if (super->animationState == 2) {
+                        super->action = 6;
+                        super->flags &= ~ENT_COLLIDE;
+                        InitializeAnimation(super, 0xe);
+                        sub_08030F00(this);
+                    }
+                }
+                break;
+            }
+    }
+}
+
+void Eyegore_Action6(EyegoreEntity* this) {
+    GetNextFrame(super);
+    if ((super->frame & 0x80) != 0) {
+        super->action = 2;
+    }
+}
+
+void sub_08030E3C(EyegoreEntity* this) {
+    super->frame = 0;
+    super->zVelocity = 0x11800;
+    EnqueueSFX(SFX_157);
+}
+
+void sub_08030E58(EyegoreEntity* this) {
+    super->action = 4;
+    this->unk_7a = 0;
+    this->unk_79 = 0;
+    this->unk_78 = 0;
+}
+
+void sub_08030E70(EyegoreEntity* this) {
+    sub_08030E80(this);
+    EnqueueSFX(SFX_158);
+}
+
+const s8 gUnk_080CE2C0[] = {
+    // This has to be before gUnk_080CE2F4.
```

Comment:
Is this the correct way to place this array as it is used in multiple functions, but placed in the data section before `gUnk_080CE2F4`?